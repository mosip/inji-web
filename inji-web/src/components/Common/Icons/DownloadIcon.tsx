import React from "react";

export function DownloadIcon({ gradient = false, style, ...props }: { gradient?: boolean, style?: React.CSSProperties, testId: string} & React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      style={style}
      aria-label={"icon-download"}
      data-testid={props.testId}
      {...props}
    >
      {gradient && (
        <defs>
          <linearGradient id="downloadGradient" x1="0" y1="0" x2="24" y2="24" gradientUnits="userSpaceOnUse">
            <stop stopColor="#6a11cb"/>
            <stop offset="1" stopColor="#2575fc"/>
          </linearGradient>
        </defs>
      )}
      <path
        d="M12 15.7885L7.73075 11.5192L8.78475 10.4348L11.25 12.9V4.5H12.75V12.9L15.2153 10.4348L16.2693 11.5192L12 15.7885ZM6.30775 19.5C5.80258 19.5 5.375 19.325 5.025 18.975C4.675 18.625 4.5 18.1974 4.5 17.6923V14.9808H6V17.6923C6 17.7692 6.03208 17.8398 6.09625 17.9038C6.16025 17.9679 6.23075 18 6.30775 18H17.6923C17.7692 18 17.8398 17.9679 17.9038 17.9038C17.9679 17.8398 18 17.7692 18 17.6923V14.9808H19.5V17.6923C19.5 18.1974 19.325 18.625 18.975 18.975C18.625 19.325 18.1974 19.5 17.6923 19.5H6.30775Z"
        fill={gradient ? "url(#downloadGradient)" : "currentColor"}
      />
    </svg>
  );
}