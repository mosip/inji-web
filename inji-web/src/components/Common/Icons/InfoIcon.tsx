import React from 'react';
import InfoIconSrc from '../../../assets/InfoIcon.svg';

interface InfoIconProps {
  className?: string;
  testId?: string;
  alt?: string;
}

export const InfoIcon: React.FC<InfoIconProps> = ({
  className = '',
  testId = 'icon-info',
  alt = 'Info'
}) => {
  return (
    <img
      data-testid={testId}
      src={InfoIconSrc}
      alt={alt}
      className={className}
    />
  );
};