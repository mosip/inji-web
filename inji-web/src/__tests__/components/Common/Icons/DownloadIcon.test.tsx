import {render} from '@testing-library/react';
import '@testing-library/jest-dom';
import {DownloadIcon} from '../../../../components/Common/Icons/DownloadIcon';

describe('DownloadIcon Component', () => {
  const defaultProps = {
    testId: 'icon-download'
  };

  const renderDownloadIcon = (props = {}) => {
    return render(<DownloadIcon {...defaultProps} {...props} />);
  };

  it('matches snapshot with default props (non-gradient)', () => {
    const { asFragment } = renderDownloadIcon();
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with gradient version', () => {
    const { asFragment } = renderDownloadIcon({ gradient: true });
    expect(asFragment()).toMatchSnapshot();
  });
});