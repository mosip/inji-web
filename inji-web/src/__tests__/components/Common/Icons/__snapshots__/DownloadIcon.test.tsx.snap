// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadIcon Component matches snapshot with default props (non-gradient) 1`] = `
<DocumentFragment>
  <svg
    aria-label="icon-download"
    data-testid="icon-download"
    fill="currentColor"
    height="24"
    testid="icon-download"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 15.7885L7.73075 11.5192L8.78475 10.4348L11.25 12.9V4.5H12.75V12.9L15.2153 10.4348L16.2693 11.5192L12 15.7885ZM6.30775 19.5C5.80258 19.5 5.375 19.325 5.025 18.975C4.675 18.625 4.5 18.1974 4.5 17.6923V14.9808H6V17.6923C6 17.7692 6.03208 17.8398 6.09625 17.9038C6.16025 17.9679 6.23075 18 6.30775 18H17.6923C17.7692 18 17.8398 17.9679 17.9038 17.9038C17.9679 17.8398 18 17.7692 18 17.6923V14.9808H19.5V17.6923C19.5 18.1974 19.325 18.625 18.975 18.975C18.625 19.325 18.1974 19.5 17.6923 19.5H6.30775Z"
    />
  </svg>
</DocumentFragment>
`;

exports[`DownloadIcon Component matches snapshot with gradient version 1`] = `
<DocumentFragment>
  <svg
    fill="none"
    height="18"
    viewBox="0 0 18 18"
    width="18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9 13.5462L3.8769 8.4231L5.1417 7.1217L8.1 10.08V0H9.9V10.08L12.8583 7.1217L14.1231 8.4231L9 13.5462ZM2.1693 18C1.5631 18 1.05 17.79 0.63 17.37C0.21 16.95 0 16.4369 0 15.8307V12.5769H1.8V15.8307C1.8 15.9231 1.8385 16.0077 1.9155 16.0845C1.9923 16.1615 2.0769 16.2 2.1693 16.2H15.8307C15.9231 16.2 16.0077 16.1615 16.0845 16.0845C16.1615 16.0077 16.2 15.9231 16.2 15.8307V12.5769H18V15.8307C18 16.4369 17.79 16.95 17.37 17.37C16.95 17.79 16.4369 18 15.8307 18H2.1693Z"
      fill="url(#paint0_linear_862_1366)"
    />
    <defs>
      <lineargradient
        gradientUnits="userSpaceOnUse"
        id="paint0_linear_862_1366"
        x1="0"
        x2="18"
        y1="9"
        y2="9"
      >
        <stop
          stop-color="#FF5300"
        />
        <stop
          offset="0.16"
          stop-color="#FB5103"
        />
        <stop
          offset="0.31"
          stop-color="#F04C0F"
        />
        <stop
          offset="0.46"
          stop-color="#DE4322"
        />
        <stop
          offset="0.61"
          stop-color="#C5363C"
        />
        <stop
          offset="0.75"
          stop-color="#A4265F"
        />
        <stop
          offset="0.9"
          stop-color="#7C1389"
        />
        <stop
          offset="1"
          stop-color="#5B03AD"
        />
      </lineargradient>
    </defs>
  </svg>
</DocumentFragment>
`;
